lib/src/main/scala/com/worksap/nlp/uzushio/lib/html/ParagraphExtractor.scala

```diff
@@ -10,16 +10,17 @@ import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
 /** Handler to segment text into paragraphs.
-  *
-  * The delimiter is used to indicate the start/end of paragraphs. It always
-  * locates the start of a line, but may contain additional text after it.
-  */
+*
+* The delimiter is used to indicate the start/end of paragraphs. It always
+* locates the start of a line, but may contain additional text after it.
+*/
 class ParagraphExtractor(
     val paragraphs: ArrayBuffer[String]
 ) extends DefaultHandler {
   private var ignoreLevel = 0
   private val writer = new StringBuilder()
   private var tag_path = mutable.Stack[String]()
+  private var per_tag_path_str = ""
```

Comment:
Does this need to be a class field?

Reply:
We need to maintain this value so when encountering a new tag, we can push the paragraph of the last tag into the result. (It is also possible but complicated by doing multiple time of `tag_path.pop()`, since we need to check again whether the tag poped by tag_path is in `blockTags`  )

Reply:
I see, gotcha!